transform_migrate.py

```diff
@@ -0,0 +1,366 @@
+from glue import sales_val_flagging as flg
+import subprocess as sp
+
+# Set working dir to manual_update, standardize yaml and src locations
+root = sp.getoutput("git rev-parse --show-toplevel")
+os.chdir(os.path.join(root))
+
+
+def write_dfs_to_s3(dfs, bucket, table):
+    """
+    Writes dicctionary of dfs to bucket
+    """
+
+    for df_name, df in dfs.items():
+        file_path = f"{bucket}/sale/{table}/{df_name}.parquet"
+
+        wr.s3.to_parquet(df=df, path=file_path, index=False)
+
+
+def read_parquets_to_dfs(prefix, table):
+    """
+    Reads all parquet files from a specified S3 path into separate pandas DataFrames,
+    names them with a given prefix, and returns them in a dictionary.
+
+    Parameters:
+    - prefix: The prefix to prepend to each DataFrame name.
+    - s3_path: The S3 path where the parquet files are stored. Should be in the format "s3://bucket-name/path/to/files".
+
+    Returns:
+    A dictionary of DataFrames keyed by their names.
+    """
+    # List all parquet files in the specified S3 path
+    parquet_files = wr.s3.list_objects(
+        os.path.join(os.getenv("AWS_S3_WAREHOUSE_BUCKET"), "sale", table),
+        suffix=".parquet",
+    )
+
+    # Initialize a dictionary to hold the dataframes
+    dfs = {}
+
+    # Loop through the parquet files, read each into a DataFrame, and store it in the dictionary
+    for file in parquet_files:
+        # Extract a meaningful name part from the file path for use in the DataFrame variable name
+        name_part = file.split("/")[-1].split(".")[
+            0
+        ]  # Adjust this as necessary based on your file naming convention
+        df_name = f"{prefix}{name_part}"
+
+        # Read the parquet file into a DataFrame
+        df = wr.s3.read_parquet(file)
+
+        # Store the DataFrame in the dictionary with the constructed name
+        dfs[df_name] = df
+
+    return dfs
+
+
+# - - - - - -
+# Adjust flag tables
+# - - - - - -
+
+# Ingest sale.flag data
+dfs_sale_flag = read_parquets_to_dfs("", "flag")
+
+# OG initial run
+# 'res_og_mansueto', 'condos_og_mansueto'
+dfs_sale_flag["2024-01-19_18:46-clever-boni"]["group"] = dfs_sale_flag[
+    "2024-01-19_18:46-clever-boni"
+]["group"].apply(
+    lambda value: value + "-res_og_mansueto"
+    if value.count("_") == 2
+    else (value + "-condos_og_mansueto" if value.count("_") == 1 else value)
+)
+
+# Update res run
+dfs_sale_flag["2024-01-29_14:40-pensive-rina"]["group"] = dfs_sale_flag[
+    "2024-01-29_14:40-pensive-rina"
+]["group"].apply(
+    lambda value: value + "-res_single_family"
+    if "40_years" in value
+    else (value + "-res_multi_family" if "20_years" in value else value)
+)
+
+# Update new condos run
+dfs_sale_flag["2024-02-01_12:24-nifty-tayun"]["group"] = (
+    dfs_sale_flag["2024-02-01_12:24-nifty-tayun"]["group"] + "-condos"
+)
+
+write_dfs_to_s3(dfs_sale_flag, os.getenv("AWS_TEST_ARCH_CHANGE_BUCKET"), "flag")
+
+# - - - - - -
+# Adjust parameter tables
+# - - - - - -
```

Comment:
In these parameter tables, the string type allowed me to have some flexibility in imputing prior run data into the structure that all of the new runs will have. For example, currently in the `stat_groups` columns, there is a long list of json string data written which contains bin information, current vs res, etc. For this og run, I impute this string:

``` python
{
        "og_mansueto": {
            "res": {"columns": ["rolling_window", "township_code", "class"]},
            "condos": {"columns": ["township_code", "rolling_window"]},
        }
    }
```

The json structure is different that the new existing structure for this column, so I decided to just simplify it to contain the relevant information. The drawback here is that the full json structure isn't consistent accross different run_id's. This can be observed in the `ci_model_sales_val_89_architecture_change_for_variable_methodology_sale.parameter` table. One alternative to the current method of "only putting what is necessary" would be to have the existing structure for every run_id and leaving lots of key values blank. You'll have to let me know what you think.

Reply:
Oooh I like this idea. I'll see if I can draft a prototype.